app/controllers/api/groups_controller.rb

```diff
@@ -281,6 +281,30 @@ def update_marking_state
       end
     end
 
+    def add_tag
+      tag = Tag.find_by(id: params[:tag_id])
```

Comment:
Tags are assignment-specific. So, you should find the grouping first, so that you can query to find tags associated only with the assignment corresponding to the grouping (rather than an arbitrary tag from a potentially different assignment).